### ISSUE TITLE
Missing Semantic Label

### ISSUE THREAD

**Issue by getBoolean** [woltapp/wolt_modal_sheet#148]
## Bug report

**Describe the bug**
Using [accessibility_tools](https://pub.dev/packages/accessibility_tools) package to check for missing semantic labels, I found that `WoltModalSheet.show` does not give a semantic label for the dismissible GestureDetector.

![flutter_boolean_template_kqm6AHtW0a](https://github.com/woltapp/wolt_modal_sheet/assets/19920697/d3ee56e6-e427-4aef-8b85-94302f73896e)

### Steps to reproduce

1. Using `WoltModalSheet.show`

<details>
<summary>
Code: (click to expand)
</summary>

```dart
abstract class HumanReadableEnum {
  const HumanReadableEnum();

  String get humanName;
}

enum ThemeType implements HumanReadableEnum {
  light('Light'),
  dark('Dark'),
  system('System');

  const ThemeType(this.humanName);

  @override
  final String humanName;
}

Future<ReturnT?> showOptionsMenu<ReturnT, OptionT extends HumanReadableEnum>(
  BuildContext context, {
  required OptionT current,
  required List<OptionT> options,
  required String title,
}) async {
  return await WoltModalSheet.show<ReturnT>(
    context: context,
    useRootNavigator: true,
    pageListBuilder: (BuildContext context) {
      final theme = Theme.of(context);
      return [
        SliverWoltModalSheetPage(
          topBarTitle: Center(
            child: Text(
              title,
              style: theme.textTheme.titleMedium,
            ),
          ),
          isTopBarLayerAlwaysVisible: true,
          leadingNavBarWidget: const Padding(
            padding: EdgeInsetsDirectional.all(16.0),
            child: CloseButton(),
          ),
          mainContentSlivers: [
            SliverList(
              delegate: SliverChildListDelegate([
                for (final eachOption in options)
                  ListTile(
                    title: Text(
                      eachOption.humanName,
                      style: current == eachOption
                          ? const TextStyle(
                              fontWeight: FontWeight.bold,
                            )
                          : null,
                    ),
                    onTap: () {
                      Navigator.of(context).pop(eachOption);
                    },
                    trailing:
                        current == eachOption ? const Icon(Icons.check) : null,
                  ),
              ]),
            ),
          ],
        ),
      ];
    },
  );
}
```

</details>

### Expected behavior

It should be possible to add a semantic label to the dismissable `GestureDetector`, or exclude it from semantics if that information isn't important to a user using TalkBack/VoiceOver

---

## Additional context

The problem seems to be with the `GestureDetector` in `wolt_modal_sheet.dart` at line 286

```dart
GestureDetector(
  behavior: HitTestBehavior.opaque,
  onTap: () {
    if (widget.route.barrierDismissible) {
      final onModalDismissedWithBarrierTap =
          widget.onModalDismissedWithBarrierTap;
      if (onModalDismissedWithBarrierTap != null) {
        onModalDismissedWithBarrierTap();
      } else {
        Navigator.of(context).pop();
      }
    }
  },
  child: const SizedBox.expand(),
)
```

---

**Comment by ulusoyca** [comment]
@getBoolean Thanks for the detailed issue. This is an easy fix and should be high priority. Added to the backlog of next release.

**Comment by ulusoyca** [comment]
@getBoolean Can you check this PR? I believe Furkan's contribution will fix this. https://github.com/woltapp/wolt_modal_sheet/pull/150
